mops-torch/src/hpe.cpp

```diff
@@ -38,15 +43,25 @@ torch::Tensor HomogeneousPolynomialEvaluation::forward(
         });
     } else if (A.device().is_cuda()) {
 
+#ifndef MOPS_CUDA_ENABLED
+        C10_THROW_ERROR(ValueError, "MOPS was not compiled with CUDA support " + A.device().str());
+#else
+        c10::cuda::CUDAGuard deviceGuard{A.device()};
```

Comment:
What does this `deviceGuard` do? I see that it's not being used explicitly

Reply:
it sets the current CUDA device to be the same one as A.device()

Reply:
There's no easy way that I can see for us to test whether a kernel has launched on a specific stream. We can probably do this with the CUDA API but that seems a bit overkill.